### PR TITLE
Implements input validation for the AplicationId

### PR DESCRIPTION
**Pull request by andreban**
- Implements validation and tests
- Improves some error messages on the CLI

### CODE REVIEW

File: packages/core/src/spec/lib/utilSpec.ts [hunk 1]
```diff
@@ -137,4 +137,34 @@ describe('util', () => {
...
+  describe('#validatePackageId', () => {
+    it('returns true for valid packages', () => {
+      expect(util.validatePackageId('com.pwa_directory.appspot.com')).toBeTrue();
+      expect(util.validatePackageId('com.pwa1directory.appspot.com')).toBeTrue();
+    });
+
+    it('returns false for packages with invalid characters', () => {
+      expect(util.validatePackageId('com.pwa-directory.appspot.com')).toBeFalse();
+      expect(util.validatePackageId('com.pwa@directory.appspot.com')).toBeFalse();
+      expect(util.validatePackageId('com.pwa*directory.appspot.com')).toBeFalse();
```
Comment (PEConn) [inline packages/core/src/spec/lib/utilSpec.ts:150]: Maybe it's worth adding an example with some weird unicode character (maybe a unicode character that looks like a full stop but isn't).

File: packages/core/src/spec/lib/utilSpec.ts [hunk 2]
```diff
@@ -137,4 +137,34 @@ describe('util', () => {
...
+
+    it('packages with less than 2 sections return false', () => {
+      expect(util.validatePackageId('com')).toBeFalse();
+      expect(util.validatePackageId('')).toBeFalse();
+    });
+
+    it('packages starting with non-letters return false', () => {
+      expect(util.validatePackageId('com.1char.twa')).toBeFalse();
+      expect(util.validatePackageId('1com.char.twa')).toBeFalse();
+      expect(util.validatePackageId('com.char.1twa')).toBeFalse();
```
Comment (PEConn) [inline packages/core/src/spec/lib/utilSpec.ts:167]: One that starts with an underscore?

File: packages/cli/src/lib/cmds/init.ts [hunk 3]
```diff
@@ -91,19 +91,26 @@ async function confirmTwaConfig(twaManifest: TwaManifest): Promise<TwaManifest>
...
-      validate: notEmpty,
+      validate: async (input): Promise<boolean> =>
+        notEmpty(input, 'KeyStore location cannot be empty'),
     }, {
       name: 'keyAlias',
       type: 'input',
       message: 'Key name:',
       default: twaManifest.signingKey.alias,
-      validate: notEmpty,
+      validate: async (input): Promise<boolean> => notEmpty(input, 'Key alias cannot be empty'),
```
Comment (PEConn) [inline packages/cli/src/lib/cmds/init.ts:113]: Could you just pass the name to `notEmpty` and have `notEmpty` take care of the "cannot be empty" part? (As it is, for half of these you say "cannot be empty" and the other half you have "can't be empty".

File: packages/cli/src/spec/inputHelpersSpec.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    });
+  });
+
+  describe('#validateKeyPassword', () => {
+    it('throws Error for empty string', async () => {
+      await expectAsync(inputHelpers.validateKeyPassword('')).toBeRejectedWithError();
+      await expectAsync(inputHelpers.validateKeyPassword(' ')).toBeRejectedWithError();
+    });
+
+    it('throws Error for empty string', async () => {
```
Comment (PEConn) [inline packages/cli/src/spec/inputHelpersSpec.ts:37]: description is wrong.

File: packages/cli/src/spec/inputHelpersSpec.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      await expectAsync(inputHelpers.validateKeyPassword('a')).toBeRejectedWithError();
+      await expectAsync(inputHelpers.validateKeyPassword('abc')).toBeRejectedWithError();
+      await expectAsync(inputHelpers.validateKeyPassword('abcde')).toBeRejectedWithError();
+      await expectAsync(inputHelpers.validateKeyPassword('abcde ')).toBeRejectedWithError();
+    });
+
+    it('throws Error for empty string', async () => {
```
Comment (PEConn) [inline packages/cli/src/spec/inputHelpersSpec.ts:44]: here as well

File: packages/core/src/spec/lib/utilSpec.ts (hunk 2 above)
Comment (andreban) [reply packages/core/src/spec/lib/utilSpec.ts:167]: `\u2024` (`․`) should do it.

File: packages/cli/src/spec/inputHelpersSpec.ts (hunk 4 above)
Comment (andreban) [reply packages/cli/src/spec/inputHelpersSpec.ts]: done

File: packages/cli/src/spec/inputHelpersSpec.ts (hunk 5 above)
Comment (andreban) [reply packages/cli/src/spec/inputHelpersSpec.ts]: done

File: packages/cli/src/lib/cmds/init.ts (hunk 3 above)
Comment (andreban) [reply packages/cli/src/lib/cmds/init.ts]: done

File: packages/core/src/spec/lib/utilSpec.ts (hunk 2 above)
Comment (andreban) [reply packages/core/src/spec/lib/utilSpec.ts:168]: done

File: packages/core/src/spec/lib/utilSpec.ts (hunk 1 above)
Comment (andreban) [reply packages/core/src/spec/lib/utilSpec.ts:150]: done

**Pull request merged**